specs/release.md

```diff
@@ -0,0 +1,469 @@
+
+### Hotfix Workflow
+
+Critical production issues requiring immediate deployment MAY bypass the standard pull request review process through an automated PR-based hotfix workflow that deploys to development on every push and requires explicit approval for production deployment.
+
+**Hotfix workflow MUST be PR-based with automatic CI/CD triggers.** The workflow automates hotfix branch creation, continuous dev deployment on push, and production deployment approval through GitHub Environment protection. This approach allows rapid iteration (push multiple fixes without manual workflow re-runs) while maintaining safety gates for production.
+
+#### Automated Hotfix Workflow Requirements
+
+The `.github/workflows/hotfix-create.yml` and `.github/workflows/hotfix-deploy.yml` workflows MUST implement the following automation:
+
+**Workflow 1: Hotfix Branch Creation (`hotfix-create.yml`)**
+
+Triggered via workflow dispatch to create the hotfix branch from current production commit.
+
+**Inputs:**
+- `issue_description` (required): Brief description of the production issue
+- `severity` (required): Critical (auth/data loss/security) or High (degraded performance)
+
+**Steps:**
+1. Query GitHub Deployments API for environment "production"
+2. Extract commit SHA from most recent successful deployment
+3. Create branch `hotfix/{timestamp}-{issue-slug}` from production commit
+4. Create draft pull request from hotfix branch to `main`:
+   - PR title: `[Hotfix] {issue_description}`
+   - PR body: Severity, production commit SHA, instructions for developer
+   - Labels: `hotfix`, `draft`
+5. Output PR URL and hotfix branch name
+
+**Workflow 2: Hotfix CI/CD Pipeline (`hotfix-deploy.yml`)**
+
+Triggered automatically on every push to `hotfix/*` branches and on PR approval.
+
+**Trigger conditions:**
+```yaml
+on:
+  push:
+    branches:
+      - 'hotfix/**'
+  pull_request:
+    types: [opened, synchronize, ready_for_review]
+    branches:
+      - main
+  pull_request_review:
+    types: [submitted]
+```
+
+**Pipeline stages:**
+
+1. **Run Tests** (on every push to hotfix/\* branch)
+   - Check out hotfix branch
+   - Run type checking (MUST pass)
+   - Run unit tests (MUST pass)
+   - Run E2E tests (MUST pass)
+   - Verify coverage meets thresholds
+   - Fail workflow if any test fails (blocks dev deployment)
+   - Update PR status check (tests passing/failing)
+
+2. **Deploy to Development** (on every push after tests pass)
+   - Trigger internal workflow call to `deploy.yml` with:
+     - `branch`: hotfix branch name
+     - `environment`: development
+   - Wait for deployment completion
+   - Verify health check passes (automatic rollback on failure)
+   - Post PR comment with development URL and deployment status
+
+3. **Deploy to Production** (on PR approval only)
+   - Requires PR marked "Ready for review" (not draft)
+   - Requires manual approval from authorized maintainer via PR review
```

Comment:
This is overkill because right now with a single maintainer we have nobody to approves PR's. We just want approval in so far as we don't want to accidentally deploy to prod. Someone has to consciously approve that deploy but not via PR review.